framework/src/postprocessors/SideDiffusiveFluxIntegral.C

```diff
@@ -60,11 +62,36 @@ SideDiffusiveFluxIntegralTempl<is_ad>::computeQpIntegral()
     const auto & grad_u = MetaPhysicL::raw_value(_fv_variable->adGradSln(*fi));
 
     // FIXME Get the diffusion coefficient on the face, see #16809
-    return -MetaPhysicL::raw_value(_diffusion_coef[_qp]) * grad_u * _normals[_qp];
+    return -diffusivity_gradient_product(grad_u, MetaPhysicL::raw_value(_diffusion_coef[_qp])) *
+           _normals[_qp];
   }
   else
-    return -MetaPhysicL::raw_value(_diffusion_coef[_qp]) * _grad_u[_qp] * _normals[_qp];
+    return -diffusivity_gradient_product(_grad_u[_qp],
+                                         MetaPhysicL::raw_value(_diffusion_coef[_qp])) *
+           _normals[_qp];
 }
 
-template class SideDiffusiveFluxIntegralTempl<false>;
-template class SideDiffusiveFluxIntegralTempl<true>;
+template <bool is_ad, typename T>
+RealVectorValue
+SideDiffusiveFluxIntegralTempl<is_ad, T>::diffusivity_gradient_product(RealVectorValue grad_u,
+                                                                       Real diffusivity)
+{
+  return grad_u * diffusivity;
+}
+
+template <bool is_ad, typename T>
+RealVectorValue
+SideDiffusiveFluxIntegralTempl<is_ad, T>::diffusivity_gradient_product(RealVectorValue grad_u,
+                                                                       RealVectorValue diffusivity)
+{
+  RealVectorValue d_grad_u = grad_u;
+  for (unsigned int i = 0; i < LIBMESH_DIM; i++)
+    d_grad_u *= diffusivity(i);
```

Comment:
This is not a component-wise multiplication. Isn't that what you want? What you are doing now will yield this vector:

(grad_u0 * d0 * d1 * d2, grad_u1 * d0 * d1 * d2, grad_u2 * d0 * d1 * d2)

Reply:
good catch. Should be d_grad_u(i)